### ISSUE TITLE
Error in cancel command implementation

### ISSUE THREAD

**Issue by rntz** [knausj85/knausj_talon#559]
```
2021-08-07 20:21:50    IO 
||||||||###o#v###e#####o###################|| 458.325 (viterbi)

||||||||###################################|| 456.474
||||||||###o#n###e#########################|| 454.019
||||||||###a#####i#r#######################|| 453.536
||||||||###o#n###e##|##o###################|| 452.781
  [reject]

2021-08-07 20:21:51    IO DECODED 
2021-08-07 20:21:51 ERROR cb error topic="pre:phrase" cb=<function pre_phrase at 0x7f2608cbb310>
   25:       lib/python3.9/threading.py:912* # cron thread
   24:       lib/python3.9/threading.py:954* 
   23:       lib/python3.9/threading.py:892* 
   22:                    talon/cron.py:155| 
   21:                    talon/cron.py:103| 
   20:          talon/scripting/rctx.py:233| # 'cron' main:<lambda>()
   19:                     talon/vad.py:21 | 
   18: talon/scripting/speech_system.py:331| 
   17:             talon/engines/w2l.py:931| 
   16:      talon/scripting/dispatch.py:105| 
   15:      talon/scripting/dispatch.py:144| 
   14:      talon/scripting/dispatch.py:133| 
   13:          talon/scripting/rctx.py:233| # 'phrase' main:_redispatch()
   12: talon/scripting/speech_system.py:61 | 
   11:      talon/scripting/dispatch.py:105| 
   10:      talon/scripting/dispatch.py:144| 
    9:      talon/scripting/dispatch.py:133| 
    8:          talon/scripting/rctx.py:233| # 'phrase' main:engine_event()
    7: talon/scripting/speech_system.py:387| 
    6:      talon/scripting/dispatch.py:105| 
    5:      talon/scripting/dispatch.py:143| 
    4:      talon/scripting/dispatch.py:113| 
    3: ------------------------------------# cron thread
    2:          talon/scripting/rctx.py:233| # 'pre:phrase' user.knausj.code.cancel:pre_phrase()
    1:       user/knausj/code/cancel.py:9  | before, after = d["text"][:-n], d['text'][-n:]
KeyError: 'text'
```

Probably just need to check whether the "text" key is present or not. Not sure though, could ask aegis.

**Comment by AndreasArvidsson** [comment]
@rntz I believe this is when you making just noises as that actually triggers pre:phrase but without any text. Just check if the key is there or add a try.

My own solution
https://github.com/AndreasArvidsson/andreas-talon/blob/master/misc/abort/abort.py

**Comment by rntz** [comment]
Fixed by #575
